test/fw/bin/pbs_benchpress

```diff
@@ -507,13 +530,14 @@ if __name__ == '__main__':
                 sys.exit(1)
             plugins += (_plugin(),)
     test_regex = r'(^(?:[\w]+|^)Test|pbs_|^test_[\(]*)'
-    os.environ['NOSE_TESTMATCH'] = test_regex
-    if nosedebug:
-        os.environ['NOSE_VERBOSE'] = '7'
-        if outfile:
-            os.environ['NOSE_DEBUG_LOG'] = outfile
-    else:
-        os.environ['NOSE_VERBOSE'] = '2'
-    if stoponfail:
-        os.environ['NOSE_STOP'] = '1'
-    nose.main(defaultTest=tests, argv=[sys.argv[0]], plugins=plugins)
+    nose_config = PTLNoseConfig(
+        log_format=fmt,
+        outfile_handler=outfile_hdlr,
+        plugins=PluginManager(plugins=plugins),
+        testMatchPat=test_regex,
```

Comment:
I didn't really find any documentation for 'testMatchPat', or much documentation for invoking nose.config.Config's constructor. I don't know a lot about nose, is it actually recommended to invoke nose.config.Config's constructor directly? If these are internal parameters then they might change and we might not want to use them.

Reply:
@ericpershey this is my only outstanding comment

Reply:
Commit 7f0124ba44964be4bafcf17637e913380a9875aa removes the use of `testMatchPat` and instead sets the `NOSE_TESTMATCH` environment variable before creating the config object.  Please let us know if this is acceptable.

Reply:
Thanks, but I'd like @hirenvadalia to vet the nose and test loader related changes once